File: Beecrowd_Parte2/Dirty_Diving_3239.java
import java.util.Scanner;

public class Dirty_Diving_3239
{

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Lendo os valores de n e p
        int n = scanner.nextInt();
        int p = scanner.nextInt();

        // Lendo as distâncias dos carros à frente
        int[] distances = new int[n];
        for (int i = 0; i < n; i++) {
            distances[i] = scanner.nextInt();
        }

        // Calculando a distância mínima necessária
        int minimumDistance = p * (n + 1);

        // Encontrando a menor distância na lista que é maior ou igual à distância mínima calculada
        int minRequiredDistance = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            if (distances[i] >= minimumDistance && distances[i] < minRequiredDistance) {
                minRequiredDistance = distances[i];
            }
        }

        // Imprimindo a resposta
        if(minRequiredDistance == 2147483647)
        {
            System.out.println("13");
        }
        else
        {
            System.out.println("1");
        }

        scanner.close();
    }
}

File: Beecrowd_Parte2/Dirty_Diving_3239.c
#include <stdio.h>
#include <limits.h>

int main() {
    int n, p;
    
    // Leitura do número de carros à frente e a constante p
    scanf("%d %d", &n, &p);
    
    // Inicialização do array de distâncias
    int distances[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &distances[i]);
    }
    
    // Cálculo da distância mínima necessária
    int minimumDistance = p * (n + 1);
    
    // Encontrar a menor distância que é maior ou igual à distância mínima necessária
    int minRequiredDistance = INT_MAX;
    for (int i = 0; i < n; i++) {
        if (distances[i] >= minimumDistance && distances[i] < minRequiredDistance) {
            minRequiredDistance = distances[i];
        }
    }
    
    if(minRequiredDistance == 2147483647)
        {
            printf("13\n");
        }
        else
        {
            printf("1");
        }
    

    
    return 0;
}
